docs/api/role-management.asciidoc

```diff
@@ -0,0 +1,23 @@
+[role="xpack"]
+[[role-management-api]]
+== Kibana Role Management API
+
+experimental[This API is *experimental* and may be changed or removed completely in a future release. The underlying mechanism of enforcing role based access control is stable, but the APIs for managing the roles are currently experimental.]
+
+The role management API allows people to manage roles that grant <<kibana-privileges>>.
+It is *not* supported to do so using the
+{ref}/security-api-roles.html[{es} Role Management APIs], and doing 
+so will likely cause {kib}'s authorization to behave unexpectedly. 
+
+[[role-management-api-endpoints]]
+=== Endpoints
+
+experimental[This API is *experimental* and may be changed or removed completely in a future release. The underlying mechanism of enforcing role based access control is stable, but the APIs for managing the roles are currently experimental.]
+
+==== Authorization
+
+To use this API, you must have at least the `manage_security` cluster privilege.
+
+include::role-management/put.asciidoc[]
+include::role-management/get.asciidoc[]
+include::role-management/delete.asciidoc[]
```

Comment:
To be consistent with how we document APIs, it would be best to put create role, get role, and delete role on separate pages.  The first page would then include an intro to the API, the warning about not using Elasticsearch Role Management API, the text about authorization, and links to each page.

[role="xpack"]
[[role-management-api]]
== Role Management API

experimental[This API is *experimental* and may be changed or removed completely in a future release. Although the underlying mechanism of enforcing role-based access control is stable, the APIs for managing the roles are currently experimental.]

The {kib} role management API enables you to control access to {kib} features and saved objects.
Managing roles that grant <<`kibana-privileges`>> using {es}'s 
{ref}/security-api-roles.html[Role Management APIs] is *not* supported. Doing 
so will likely cause {kib}'s authorization to behave unexpectedly. 

To use the {kib} role management API, you must have the `manage_security` cluster 
privilege at minimum.

* <<`role-management-api-put`>>
* <<`role-management-api-get`>>
* <<`role-management-api-delete`>>

include::role-management/put.asciidoc[]
include::role-management/get.asciidoc[]
include::role-management/delete.asciidoc[]

Reply:
Yeah... I had them this way at one point, I can definitely put them back, it felt weird having that Authorization section on every page, so I did something closer to what ES does here: https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-roles.html

Reply:
👍